Vagrantfile

```diff
@@ -4,6 +4,7 @@
 Vagrant.require_version ">= 1.8"
 
 require 'yaml'
+require './deep_merge'
```

Comment:
How about `require 'vagrant/util/deep_merge'` and then `Vagrant::Util::DeepMerge.deep_merge(config_data_dist, config_data)` ?

It seems simpler and the implementation of `deep_merge` will be supported by vagrant developers.

See https://github.com/hashicorp/vagrant/commit/4c9e6e46b2a6cf7b7949ab85e960957fdf5faf83

Reply:
Even better! Will do

Reply:
@adamof will you be able to contribute the proposed fix in the nearest days? If you don't have time, I will work deliver it

Reply:
Hey! Go ahead! I'm laptop-less at the moment :/

Reply:
Sounds good, thanks for contributing anyway!